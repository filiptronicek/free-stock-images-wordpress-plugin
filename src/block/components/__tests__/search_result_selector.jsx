import SearchResultSelector from './../search_result_selector.jsx';
import renderer from 'react-test-renderer';
import React from 'react';

import SearchController from './../../controllers/search.jsx';
import MockImageService from './../__mocks__/services/image.jsx';

it('renders with query and sample provider', () => {
    const imageService = new MockImageService()
    const allProviderNames = imageService.providers.map((provider) => provider.name)
    const searchController = new SearchController('query', allProviderNames, [])
    const tree = renderer
        .create(<SearchResultSelector searchController={searchController} imageService={imageService} />)
        .toJSON();
  expect(tree).toMatchSnapshot();
});
