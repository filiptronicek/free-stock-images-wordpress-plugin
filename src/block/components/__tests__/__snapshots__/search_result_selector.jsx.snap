// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with query and sample provider 1`] = `
Array [
  <div
    className="MuiGrid-root searchResults MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      <img
        className="image"
        onClick={[Function]}
        src="https://imgur.to/500x400"
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
    >
      
    </div>
  </div>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Load More
    </span>
  </button>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Save
    </span>
  </button>,
]
`;
